http/README.md

```diff
@@ -0,0 +1,161 @@
+
+Support for client authentication is an optional feature. It is expected that only a subsection of clients will implement it. For example, browser simply retrieving a few (elements of) web pages from IPFS probably won’t have any need to even generate a libp2p identity in the first place.
+
+The protocol defined here takes 2 RTTs to authenticate the client. It is designed to be stateless on the server side. In the first round-trip, the client obtains a (pseudo-) random value from the server, which it then signs with its host key and sends back to the server, which then issues an authentication token (acting somewhat like a cookie) which can be included on future requests.
+
+The service name is `client-auth`. For the first step, the client sends a GET request to this HTTP endpoint, and the server responds with at least 8 and up to 1024 bytes of pseudorandom data:
+
+```json
+{
+	"random": <multibase-encoded random bytes>
+}
+```
+
+In order to keep this exchange stateless, the server SHOULD 1. include the current timestamp or an expiry data and 2. a signature in that data. This allows it to check in step 2 that it actually generated that data.
+
+The client signs the data received in step 1, and sends a POST request with the following JSON object to the server:
+
+```json
+{
+  "data": <the random bytes received from the server, multibase encoded>,
+  "peer-id": <peer ID, in string reprensentation>,
+  "signature": <multibase-encoded signature>
+}
+```
+
+The server verifies the signature and issues an authentication token. In order to allow stateless operation, at the very minimum, the authentication token SHOULD contain the peer ID. It SHOULD also contain an expiry date and it MAY be bound to the client’s IP address. The token is sent in the response body.
+
+The client uses the auth token on requests that require client authentication, by setting the `libp2p-auth-token` HTTP header.
+
+## Mapping to libp2p Streams
+
+libp2p services whose service is specified as request-response protocols can use a single protocol implementation to make the service available over HTTP as well as on top of libp2p streams.
+
+The libp2p protocol identifier is `/http1.1`. After negotiating this protocol using multistream-select, nodes treat the stream as a HTTP/1.1 stream for a single HTTP request (i.e. nodes MUST NOT use request pipelining).
+
+## Outlook: Interaction with Intermediaries
+
+One of the advantages of running HTTP is that there’s widely deployed caching infrastructure (CDNs). Content-addressed data is infinitely cacheable. Assuming a properly design data transfer protocol, retrieval for CIDs could be cached by the CDN and made available via a POP (geographically) close to the user, dramatically reducing retrieval latencies.
+
+Services SHOULD specify the caching properties (if any), and set the appropriate cache headers (according to RFC 9111).
```

Comment:
Yes, in my understanding, the request method is part of the specification for a REST endpoint, isn’t it?

Reply:
Protocols will use the existing HTTP abstractions to define their logic.

This is different than rust-libp2p's [request-response protocol](https://docs.rs/libp2p/0.50.0/libp2p/request_response/index.html). Instead of a custom serialization of request/response, we adopt all of the HTTP semantics.